### PR TITLE
Support to new calls for getting CDB lists

### PR DESCRIPTION
**Pull request by druizz90**
Hi team,

This PR is for giving support to new calls added in PR [#301](https://github.com/wazuh/wazuh-api/pull/301) (API).

Best regards,

Demetrio.

### CODE REVIEW

**Comment by druizz90** [comment]
Hi,

I made changes for showing relative paths in the output:

```bash
# curl -u foo:bar -k -X GET "http://127.0.0.1:55000/lists?pretty&offset=1&limit=2&sort=-path"
{
   "error": 0,
   "data": {
      "totalItems": 2,
      "items": [
         {
            "path": "etc/lists/audit-keys",
            "items": [
               {
                  "key": "audit-wazuh-w",
                  "value": "write"
               },
               {
                  "key": "audit-wazuh-r",
                  "value": "read"
               },
               {
                  "key": "audit-wazuh-a",
                  "value": "attribute"
               },
               {
                  "key": "audit-wazuh-x",
                  "value": "execute"
               },
               {
                  "key": "audit-wazuh-c",
                  "value": "command"
               }
            ]
         },
         {
            "path": "etc/lists/amazon/aws-test",
            "items": [
               {
                  "key": "AttachLoadBalancers",
                  "value": "Autoscaling"
               },
               {
                  "key": "DetachLoadBalancers",
                  "value": "Autoscaling"
               },
               {
                  "key": "CreateSubnet",
                  "value": "Legacy Wazuh CDB"
               },
               {
                  "key": "CreateTags",
                  "value": "Legacy Wazuh CDB"
               },
               {
                  "key": "CreateVolume",
                  "value": "Legacy Wazuh CDB"
               },
               {
                  "key": "DeletePlacementGroup",
                  "value": "Legacy Wazuh CDB"
               }
            ]
         }
      ]
   }
}
```

```bash
curl -u foo:bar -k -X GET "http://127.0.0.1:55000/lists/files?pretty"                         
{
   "error": 0,
   "data": [
      {
         "path": "etc/lists/amazon",
         "name": "aws-eventnames"
      },
      {
         "path": "etc/lists/amazon",
         "name": "aws-test"
      },
      {
         "path": "etc/lists",
         "name": "audit-keys"
      },
      {
         "path": "etc/lists",
         "name": "aa-test-list"
      }
   ]
}
```

**Comment by crd1985** [comment]
We need some unit tests to go ahead.

File: framework/wazuh/cdb_list.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2015-2019, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+
+from wazuh.exception import WazuhException
```
[the remark this answers is not in the document]
Comment (druizz90) [reply framework/wazuh/cdb_list.py]: done!

File: framework/wazuh/cdb_list.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+from wazuh import common
+from os import listdir
+from os.path import isfile, isdir, join
+from wazuh.utils import sort_array, search_array
+
+
+def get_lists(path=None, offset=0, limit=common.database_limit, sort=None, search=None):
+    """
+    Get CDB lists
```
[the remark this answers is not in the document]
Comment (druizz90) [reply framework/wazuh/cdb_list.py]: done!

File: framework/wazuh/cdb_list.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    """
+
+    output = []
+
+    if path:
+        items = get_list_from_file(path)
+        output.append(items)
+    else:
+        dir_content = listdir(common.lists_path)
+        for name in dir_content:
```
[the remark this answers is not in the document]
Comment (druizz90) [reply framework/wazuh/cdb_list.py]: done!

File: framework/wazuh/cdb_list.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    output = []
+
+    try:
+        with open(file_path) as f:
+            for line in f.read().splitlines():
+                key = line.split(':')[0]
+                value = line.split(':')[1]
+                output.append({'key': key, 'value': value})
+                        
+    except Exception as e:
```
[the remark this answers is not in the document]
Comment (druizz90) [reply framework/wazuh/cdb_list.py]: done!

File: framework/wazuh/cdb_list.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    """
+    dir_content = listdir(common.lists_path)
+
+    output = []
+
+    for name in dir_content:
+        absolute_path = join(common.lists_path, name)
+        relative_path = 'etc/lists'
+
+        if (isfile(absolute_path)) and ('.cdb' not in name) \
```
[the remark this answers is not in the document]
Comment (druizz90) [reply framework/wazuh/cdb_list.py]: done!

**Comment by druizz90** [comment]
I think is necessary to add a logger for framework @crd1985

**Comment by druizz90** [comment]
I made two fix:

- `search` parameter only filter by `path` field.
- `.swp` files are skipped. If a CDB list file is opened with `vim` (for example), an error happens when `get_list_from_file` is called (it tries to open the `.swp` file).

**Comment by druizz90** [comment]
Unit tests for CDB lists were added. I think this PR is ready!

**Comment by crd1985** [comment]
> I think is necessary to add a logger for framework @crd1985
I fully agree. Could you please open an issue to keep track of this enhancement? Thx

File: framework/wazuh/cdb_list.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+
+def _check_path(path):
+    """
+    Check if path is well formed (without './' or '../')
+    :param path: Path to check
+    :return: Result of check the path (boolean)
+    """
+    regex_path = r'^(etc/lists/)[\w\.\-/]+$'
+    pattern = re.compile(regex_path)
```
Comment (crd1985) [inline framework/wazuh/cdb_list.py]: Compiling a regex is expensive. I would suggest defining regex at a module scope and compiling them only once.

**Comment by druizz90** [comment]
#### Unit tests

```bash

$ python3 -m pytest -v
================================================ test session starts =================================================
platform linux -- Python 3.6.3, pytest-4.2.0, py-1.7.0, pluggy-0.8.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /wazuh/framework, inifile:
collected 11 items

tests/test_cdb_list.py::TestCDBList::test_get_lists PASSED                                                     [  9%]
tests/test_cdb_list.py::TestCDBList::test_get_lists_limit PASSED                                               [ 18%]
tests/test_cdb_list.py::TestCDBList::test_get_lists_offset PASSED                                              [ 27%]
tests/test_cdb_list.py::TestCDBList::test_get_lists_path PASSED                                                [ 36%]
tests/test_cdb_list.py::TestCDBList::test_get_lists_search PASSED                                              [ 45%]
tests/test_cdb_list.py::TestCDBList::test_get_lists_sort PASSED                                                [ 54%]
tests/test_cdb_list.py::TestCDBList::test_get_path_lists PASSED                                                [ 63%]
tests/test_cdb_list.py::TestCDBList::test_get_path_lists_limit PASSED                                          [ 72%]
tests/test_cdb_list.py::TestCDBList::test_get_path_lists_offset PASSED                                         [ 81%]
tests/test_cdb_list.py::TestCDBList::test_get_path_lists_search PASSED                                         [ 90%]
tests/test_cdb_list.py::TestCDBList::test_get_path_lists_sort PASSED                                           [100%]

```

#### Mocha tests

```bash
$ mocha test/test_lists.js

  Lists
    GET/lists
      ✓ Request (308ms)
      ✓ Pagination (286ms)
      ✓ Retrieve all elements with limit=0 (282ms)
      ✓ Sort (289ms)
      ✓ Search (235ms)
      ✓ Filters: Path (250ms)
      ✓ Filters: Invalid filter
      ✓ Filters: Invalid filter - Extra field
    GET/lists/files
      ✓ Request (253ms)
      ✓ Pagination (273ms)
      ✓ Retrieve all elements with limit=0 (271ms)
      ✓ Sort (271ms)
      ✓ Search (308ms)
      ✓ Filters: Invalid filter
      ✓ Filters: Invalid filter - Extra field
```

**Pull request merged**